### PR TITLE
Bugfix/v3.17.x.x offline mode with permissions

### PR DESCRIPTION
**Pull request by Szczyrk**

### CODE REVIEW

File: Assets/PatchKit Patcher/Scripts/Patcher.cs [hunk 1]
```diff
@@ -456,7 +456,10 @@ private void ThreadExecution(CancellationToken cancellationToken)
                     }
                 }
 
-                EnsureSingleInstance();
+                if (_data.Value.IsOnline != null && _data.Value.IsOnline.Value)
```
Comment (witcher112) [inline Assets/PatchKit Patcher/Scripts/Patcher.cs:459]: Please introduce here a comment explaining the cause of this `if` statement.

**Review by witcher112 — CHANGES_REQUESTED** [review]

File: Assets/PatchKit Patcher/Scripts/Patcher.cs (hunk 1 above)
Comment (Szczyrk) [reply Assets/PatchKit Patcher/Scripts/Patcher.cs]: Replacing "if" with catching the UnauthorizedAccessException exception

File: Assets/PatchKit Patcher/Scripts/Patcher.cs [hunk 2]
```diff
@@ -580,6 +580,11 @@ private void EnsureSingleInstance()
                     _lockFileStream = File.Open(lockFilePath, FileMode.Append);
                     DebugLogger.Log("Lock file open success");
                 }
+                catch (UnauthorizedAccessException exception)
```
Comment (witcher112) [inline Assets/PatchKit Patcher/Scripts/Patcher.cs:583]: Please check if this exception isn't actually thrown in multiple instances situation. If it is, an if statement checking for online/offline mode might be required here (as well as an appropriate comment explaining this mechanism).

**Review by witcher112 — CHANGES_REQUESTED** [review]

File: Assets/PatchKit Patcher/Scripts/Patcher.cs (hunk 2 above)
Comment (Szczyrk) [reply Assets/PatchKit Patcher/Scripts/Patcher.cs:583]: [output_log.txt](https://github.com/patchkit-net/patchkit-patcher-unity/files/5558852/output_log.txt)

These are logs from the launch of two patchers in online mode, you can see that one of them have an error:
`2020-11-18 09:47:33.867 +01:00 [ ERROR ] <Patcher::ThreadExecution> An exception has occured
PatchKit.Unity.Patcher.MultipleInstancesException: Another instance of Patcher spotted
  at PatchKit.Unity.Patcher.Patcher.EnsureSingleInstance () [0x00000] in <filename unknown>:0 
  at PatchKit.Unity.Patcher.Patcher.ThreadExecution (CancellationToken cancellationToken) [0x00000] in <filename unknown>:0 `

So UnauthorizedAccessException doesn't catch in this situation. I hope I understood your comment correctly.

Comment (witcher112) [reply Assets/PatchKit Patcher/Scripts/Patcher.cs:583]: Did these patchers include your modification?

Comment (Szczyrk) [reply Assets/PatchKit Patcher/Scripts/Patcher.cs:583]: Yes

**Pull request merged**